tools/replay/main.cc

```diff
@@ -4,8 +4,14 @@
 #include "common/prefix.h"
 #include "tools/replay/consoleui.h"
 #include "tools/replay/replay.h"
+#include "tools/replay/util.h"
 
 int main(int argc, char *argv[]) {
+#ifdef __APPLE__
+  // With all sockets opened, we might hit the default limit of 256 on macOS
+  setFileDescriptorLimit(1024);
```

Comment:
this doesn't have anything to do with replay, should probably go in the messaging backend

Reply:
But it only affects replay process, because of extensive of sockets. Not every process needs to increase that limit, since most processes use only a couple of sockets at most. So I'd argue that it's the process responsibility to handle that.

Reply:
But it could be moved elsewhere, so each process would be able to set it up if needed. Maybe `common/util.cc`?

Reply:
Moved `set_file_descriptor_limit` to common, so its accessible to other tools/processes if ever needed.